core/src/main/java/org/apache/carbondata/core/indexstore/ExtendedBlocklet.java

```diff
@@ -161,30 +176,35 @@ public void setColumnSchema(List<ColumnSchema> columnSchema) {
    * @param uniqueLocation
    * @throws IOException
    */
-  public void serializeData(DataOutput out, Map<String, Short> uniqueLocation)
+  public void serializeData(DataOutput out, Map<String, Short> uniqueLocation, boolean isCountJob)
       throws IOException {
     super.write(out);
-    if (dataMapUniqueId == null) {
-      out.writeBoolean(false);
+    if (isCountJob) {
```

Comment:
change this code to if(isCountJob) {//your code, return; }

Reply:
this is a if else block, return is not required. i have added return to the read method instead.